Classes/Neos/Neos/Ui/Controller/BackendServiceController.php

```diff
@@ -262,8 +272,35 @@ public function changeBaseWorkspaceAction($targetWorkspaceName)
             $updateWorkspaceInfo->setWorkspace($userWorkspace);
             $this->feedbackCollection->add($updateWorkspaceInfo);
 
-            $reloadDocument = new ReloadDocument();
-            $this->feedbackCollection->add($reloadDocument);
+            // Construct base workspace context
+            $originalContext = $documentNode->getContext();
+            $contextProperties = $documentNode->getContext()->getProperties();
+            $contextProperties['workspaceName'] = $targetWorkspaceName;
+            $contentContext = $this->contextFactory->create($contextProperties);
+
+            // If current document node doesn't exist in the base workspace, traverse its parents to find the one that exists
+            $redirectNode = $documentNode;
+            while (true) {
```

Comment:
uhm, I have to check this while(true) logic in detail; it's always hard to see that it does not end up with an endless loop ;)

Reply:
Sure, feel free to refactor if you know how to express it better. But I think it's safe.